engine/api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessState.scala

```diff
@@ -18,8 +18,15 @@ trait ProcessStateDefinitionManager {
 object ProcessState {
   import io.circe.syntax._
 
-  implicit val statusEncoder: Encoder[StateStatus] = Encoder.encodeJson.contramap(st => StateStatusCodec(st.getClass.getName, st.name).asJson)
-  implicit val statusDecoder: Decoder[StateStatus] = Decoder.decodeNone.map(st => SimpleStateStatus.Unknown) //TODO: Add decode implementation by clazz and value. At now we don't need it.
+  implicit val statusEncoder: Encoder[StateStatus] = Encoder.encodeJson.contramap(st => StateStatusCodec(st.getClass.getSimpleName, st.name).asJson)
+  implicit val statusDecoder: Decoder[StateStatus] = Decoder.decodeJson
```

Comment:
I don't think that you should invoke `json.as` inside Decoder. It is rather end-user api. You should rather do `Decoder[StateStatusCodec].map(StateStatus.codecToStateStatus)`. In your approach you will probably end with some strange exception during decoding in case of `Left(exception)`...